### ISSUE TITLE
issues about get_grads_G，get_grads_D_WAN

### ISSUE THREAD

**Issue by liang233** [goldhuang/SRGAN-PyTorch#2]
hello，I want to konw why should the code calculate the gradient of top and the bottom layer and what the grad represents？ thanks

**Comment by goldhuang** [comment]
Gradients of top and bottom layers indicate the trend of the training (if the code has bugs, the training is still healthy or it towards local/global optima). I feel it's a good debugging log to watch.

**Comment by liang233** [comment]
if gradients of top and bottom layers  are close to 0,  is the training correct ？
![default](https://user-images.githubusercontent.com/8869756/52844122-ddc63a80-313e-11e9-8ce1-d82590f9eb33.png)

**Comment by goldhuang** [comment]
Looks all right.

**Comment by liang233** [comment]
thanks
